nio/api.py

```diff
@@ -272,7 +272,11 @@ def register(
         if device_name:
             content_dict["initial_device_display_name"] = device_name
 
-        return "POST", path, Api.to_json(content_dict)
+        second_half = Api.to_json(content_dict)
+
+        first_half = '{"auth": {"type": "m.login.dummy"},'
```

Comment:
This is a bit of a weird way to do it. You can just put the auth dict into the content_dict up above.

Reply:
Other than login_raw is there somewhere else you used an auth_dict that I could use as a base? If I put it into the content dict, it doesn't work. Because it needs to be in another {} block.
Eg:
{
  "auth": {
    "example_credential": "verypoorsharedsecret",
    "session": "xxxxx",
    "type": "example.type.foo"
  },
  "device_id": "GHTYAJCE",
  "inhibit_login": false,
  "initial_device_display_name": "Jungle Phone",
  "password": "ilovebananas",
  "username": "cheeky_monkey"
}

When I add stuff into the content_dict it ends up in the bottom part, not the top part.

Reply:
The spec lists this as an example:
```python

{
  "auth": {
    "type": "example.type.foo",
    "session": "xxxxx",
    "example_credential": "verypoorsharedsecret"
  },
  "username": "cheeky_monkey",
  "password": "ilovebananas",
  "device_id": "GHTYAJCE",
  "initial_device_display_name": "Jungle Phone",
  "inhibit_login": false
}
````
Python does support nested dictionaries which will end up being what you want, no?

Reply:
When I tried the nested dictionary yesterday I was getting errors. I'll try again today; see if I can figure out where I went wrong.

Reply:
I believe my latest commit just now is more in line with the rest of your code.